packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -75,14 +75,18 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
 
     $stopPlugin(contextPath: string): PromiseLike<void> {
         this.activatedPlugins.forEach(plugin => {
-            if (plugin.stopFn) {
-                plugin.stopFn();
-            }
-
-            // dispose any objects
-            const pluginContext = plugin.pluginContext;
-            if (pluginContext) {
-                dispose(pluginContext.subscriptions);
+            try {
+                if (plugin.stopFn) {
+                    plugin.stopFn();
+                }
+
+                // dispose any objects
+                const pluginContext = plugin.pluginContext;
+                if (pluginContext) {
+                    dispose(pluginContext.subscriptions);
```

Comment:
looking at `dispose` implementation, it does not care about exceptions either

Reply:
maybe pluginContext.subscriptions is undefined

Reply:
I think the above problem should occur after they(plugin of vscode-java and ms-python) are loaded at the same time.
I try many times.

Reply:
Precondition：
1、install vscode-java、ms-python eg.
2、theia to be started
3、open java project in workspace
3、at browser，run ‘f5‘ again and again

So: the jdt process to increase and the oom phenomenon

Reply:
1、Does this function('stopFn()') not cause an unknown exception during the call?
2、Does each plugin have an implementation of this function('stopFn()') ?

Reply:
> Does each plugin have an implementation of this function('stopFn()') ?

It does not matter:
- when you reload a window, the whole process tree gets killed with java LS process as well
- when you load a new window, a new java LS process is started

Reply:
According to the implementation in the source code, I see that the JDT stop process is to stop the JDT parent process, then the parent PID of the JDT process will become 1, and finally destroyed by the operating system.

Reply:
disposable-util.ts
I have encountered this kind of mistake：
(1)
root ERROR Uncaught Exception:  TypeError: d.dispose is not a function
root ERROR TypeError: d.dispose is not a function
(2)

Reply:
It would be interesting to investigate/debug what `d` stands for. Where does it coming from? if `d` can be something else except `Disposable` then a proper fix would be handle it properly.

Reply:
root ERROR [hosted-plugin: 1777] TypeError: Cannot read property 'terminalName' of undefined
    at i.debug.onDidTerminateDebugSession.t (/home/user/.tmp/vscode-unpacked/vscjava.vscode-maven-0.16.2-huawei.vsix/extension/dist/extension.js:126:7339)
    at /opt/theia/packages/core/lib/common/event.js:161:33
    at CallbackList.invoke (/opt/theia/packages/core/lib/common/event.js:176:39)
    at Emitter.fire (/opt/theia/packages/core/lib/common/event.js:264:29)
    at DebugExtImpl.<anonymous> (/opt/theia/packages/plugin-ext/lib/plugin/node/debug/debug.js:214:60)
    at step (/opt/theia/packages/plugin-ext/lib/plugin/node/debug/debug.js:32:23)
    at Object.next (/opt/theia/packages/plugin-ext/lib/plugin/node/debug/debug.js:13:53)
    at /opt/theia/packages/plugin-ext/lib/plugin/node/debug/debug.js:7:71
    at new <anonymous> (/home/user/.tmp/vscode-unpacked/shengchen.vscode-checkstyle-1.0.2-huawei.vsix/extension/dist/extension.js:1:103420)
    at __awaiter (/opt/theia/packages/plugin-ext/lib/plugin/node/debug/debug.js:3:12)

root ERROR [hosted-plugin: 1031] (node:1031) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 48)
(node:1031) UnhandledPromiseRejectionWarning: Error: Unknown actor WorkspaceMain
    at RPCProtocolImpl.doInvokeHandler (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:200:19)
    at RPCProtocolImpl.invokeHandler (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:192:41)
    at RPCProtocolImpl.receiveRequest (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:154:45)
    at RPCProtocolImpl.receiveOneMessage (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:123:22)
    at /opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:55:89
    at /opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:233:17
    at /opt/theia/packages/core/lib/common/event.js:161:33
    at CallbackList.invoke (/opt/theia/packages/core/lib/common/event.js:176:39)
    at Emitter.fire (/opt/theia/packages/core/lib/common/event.js:264:29)
    at ChildProcess.<anonymous> (/opt/theia/packages/plugin-ext/lib/hosted/node/hosted-plugin-process.js:117:21)

Reply:
@theia/core 0.6.0
@theia/output 0.6.0
@theia/process 0.6.0
@theia/filesystem 0.6.0
@theia/variable-resolver 0.6.0
@theia/workspace 0.6.0
@theia/languages 0.6.0
@theia/editor 0.6.0
@theia/navigator 0.6.0
@theia/markers 0.6.0
@theia/outline-view 0.6.0
@theia/monaco 0.6.0
@theia/callhierarchy 0.6.0
@theia/console 0.6.0
@theia/json 0.6.0
@theia/userstorage 0.6.0
@theia/preferences 0.6.0
@theia/terminal 0.6.0
@theia/task 0.6.0
@theia/cpp 0.6.0
@theia/debug 0.6.0
@theia/debug-nodejs 0.6.0
@theia/editor-preview 0.6.0
@theia/editorconfig 0.6.0
@theia/file-search 0.6.0
@theia/keymaps 0.6.0
@theia/getting-started 0.6.0
@theia/scm 0.6.0
@theia/git 0.6.0
@theia/merge-conflicts 0.6.0
@theia/messages 0.6.0
@theia/metrics 0.6.0
@theia/mini-browser 0.6.0
@theia/search-in-workspace 0.6.0
@theia/plugin-ext 0.6.0
@theia/plugin-ext-vscode 0.6.0
@theia/preview 0.6.0
@theia/python 0.6.0
@theia/textmate-grammars 0.6.0
@theia/typehierarchy 0.6.0
@theia/typescript 0.6.0

Reply:
java
0.44.0
Java Linting, Intellisense, formatting, refactoring, Maven/Gradle support and more...
redhat
python
2019.3.6215
Linting, Debugging (multi-threaded, remote), Intellisense, code formatting, refactoring, unit tests, snippets, and more.
ms-python
Go
0.10.2
Rich Go language support for Visual Studio Code
ms-vscode
vscode-checkstyle
1.0.2
Provide real-time feedback about Checkstyle violations and quick fix actions
shengchen
vscode-java-debug
0.15.0
A lightweight Java debugger for Visual Studio Code
vscjava
vscode-java-dependency
0.4.0
Manage Java Dependencies in VSCode
vscjava
vscode-java-test
0.16.0
Run and debug JUnit or TestNG test cases
vscjava
vscode-maven
0.16.2
Manage Maven projects, execute goals, generate project from archetype, improve user experience for Java developers.
vscjava